Assets/MRTK/Providers/XRSDK/GenericXRSDKSpatialMeshObserver.cs

```diff
@@ -73,30 +111,16 @@ private async void EnableIfLoaderBecomesActive()
         #region BaseSpatialObserver Implementation
 
         private XRMeshSubsystem meshSubsystem;
-
-        /// <summary>
-        /// Creates the XRMeshSubsystem and handles the desired startup behavior.
-        /// </summary>
-        protected override void CreateObserver()
-        {
-            if (Service == null
+        private XRMeshSubsystem MeshSubsystem => meshSubsystem != null && meshSubsystem.running
+            ? meshSubsystem :
 #if XR_MANAGEMENT_ENABLED
-                || XRGeneralSettings.Instance == null || XRGeneralSettings.Instance.Manager == null || XRGeneralSettings.Instance.Manager.activeLoader == null
-#endif // XR_MANAGEMENT_ENABLED
-                ) { return; }
-
-#if XR_MANAGEMENT_ENABLED
-            meshSubsystem = XRGeneralSettings.Instance.Manager.activeLoader.GetLoadedSubsystem<XRMeshSubsystem>();
+            meshSubsystem = IsActiveLoader ?? false
+                ? XRGeneralSettings.Instance.Manager.activeLoader.GetLoadedSubsystem<XRMeshSubsystem>()
```

Comment:
Do we need to null check here?

Reply:
Albeit somewhat indirectly, `IsActiveLoader` should ensure that all of these entries are non-null before we get to this point: https://github.com/microsoft/MixedRealityToolkit-Unity/blob/3fbcd97462914be306319ea8ca09b9817b9a7fc5/Assets/MRTK/Providers/XRSDK/GenericXRSDKSpatialMeshObserver.cs#L51-L53